### PR TITLE
lib/utils/pyhelp.c: Use mp_printf() instead of printf()

### PR DESCRIPTION
**Pull request by eerimoq**
Print all help using the same printer instead of using `printf()` for parts of the output.

### CODE REVIEW

File: py/mpprint.h [hunk 1]
```diff
@@ -39,6 +39,12 @@
 #define PF_FLAG_ADD_PERCENT       (0x100)
 #define PF_FLAG_SHOW_OCTAL_LETTER (0x200)
 
+#if MICROPY_PY_IO
+#    define MP_PRINT mp_sys_stdout_print
```
Comment (pfalcon) [inline py/mpprint.h]: Based on the contents of the file you modify, MP_PRINT would be a function.

Also, there're existing code where there's mp_sys_stdout_print vs mp_plat_print dichotomy, what about it? (Should be a separate commit if anything).

Comment (eerimoq) [inline py/mpprint.h]: Okay, something like the code below? 

mpprint.h

```
const mp_print_t *mp_default_print(void);
```

mpprint.c

```
const mp_print_t *mp_default_print(void)
{
#if MICROPY_PY_IO
    return &mp_sys_stdout_print;
#else
    return &mp_plat_print;
#endif
}
```

There are also places where there's ``stream_obj`` and ``print`` vs ``mp_plat_print``.
 For example here: https://github.com/micropython/micropython/blob/master/py/modbuiltins.c#L399-L428. How should those be handled?

There is also the alternative to define one of ``mp_plat_print`` and ``mp_sys_stdout_print`` as ``mp_print`` in the header and use that printer everywhere. I mean, should the two printers ever exist at the same time in an application?

Comment (eerimoq) [inline py/mpprint.h]: There is not a single define in the MicroPython repository starting with ``MP_PYTHON_``. The common denominator for defines seems to by that they start with ``MP_``, and often ``MP_<"filename">``. So maybe ``MP_PRINT_PRINTER``, or simply ``MP_PRINTER``?

Comment (dpgeorge) [inline py/mpprint.h]: MP_PRINTER sounds good to me, but I'd make it include the `&` in the macro definition in case it changes one day to use indirection (eg with a runtime-modifiable stdout stream).

**Comment by pfalcon** [comment]
> There is not a single define in the MicroPython repository starting with MP_PYTHON_. The common denominator for defines seems to by that they start with MP_, and often MP_<"filename">. So maybe MP_PRINT_PRINTER, or simply MP_PRINTER?

All identifiers are named by their *meaning*, so the only reason something named as "MP_<"filename">" is because filename is named by meaning too, and that's how they match - not because identifiers match to filenames, but because both match to the meaning. If we introduce a printing data structure which should be used for any Python functions' output (and not necessarily for any other purpose), it's only natural to call it MP_PYTHON_PRINTER, and confusing to call it over-generally MP_PRINTER, which may only lead to mistakes in its usage, and need to fix something later again.

So, I'd still recommend to use MP_PYTHON_PRINTER.

**Comment by eerimoq** [comment]
Do I have to change anything or can this be merged as is?

**Comment by pfalcon** [comment]
> So, I'd still recommend to use MP_PYTHON_PRINTER.

**Comment by dpgeorge** [comment]
Yes, please change to MP_PYTHON_PRINTER.

**Comment by eerimoq** [comment]
New commit available with the name change.

**Comment by dpgeorge** [comment]
Thanks, merged in f64e806f50606902323c62cf7b479922724022b6.
